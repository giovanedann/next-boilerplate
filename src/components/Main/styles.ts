import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
`

export const Logo = styled.img``

export const Title = styled.h1``

export const Description = styled.h2``

export const Illustration = styled.img``
