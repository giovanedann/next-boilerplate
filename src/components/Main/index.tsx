import { FC } from 'react'
import * as S from './styles'

interface IMain {
  title?: string
  description?: string
}

const variants = {
  animate: {
    rotate: [0, 90, 180, 270, 360],
    transition: {
      repeat: Infinity,
      duration: 5,
      ease: 'linear'
    }
  },
  hover: {
    y: 200,
    scale: 3
  }
}

const Main: FC<IMain> = ({
  title = 'Boilerplate React + NextJS',
  description = 'TypeScript, React, NextJS, Jest, Testing Library, Storybook, Framer Motion and Styled Components'
}) => (
  <S.Wrapper>
    <S.Logo
      variants={variants}
      animate="animate"
      whileHover="hover"
      src="img/icon-512.png"
      alt="white react athom with pink background"
    />
    <S.Title>{title}</S.Title>
    <S.Description> {description} </S.Description>
    <S.Illustration src="img/hero-illustration.svg" alt="Developer coding" />
  </S.Wrapper>
)

export default Main
