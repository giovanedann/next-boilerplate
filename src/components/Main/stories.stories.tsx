import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  argTypes: {
    // place your args types here (inputs of type color, radio, etc)
  }
} as ComponentMeta<typeof Main>

const Template: ComponentStory<typeof Main> = (args) => <Main {...args} />

export const Basic = Template.bind({})
Basic.args = {
  description: 'This is an example of description',
  title: 'This is an example of title'
}
