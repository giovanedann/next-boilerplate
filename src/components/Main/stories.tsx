import Main from '.'
import { Story, Meta } from '@storybook/react/types-6-0'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Default title',
    description: 'Default description'
  }
} as Meta

export const Default: Story = (args) => <Main {...args} />

export const Basic: Story = (args) => <Main {...args} />

Basic.args = {
  title: 'Storybook Basic Story title',
  description: 'Storybook Basic Story description'
}
