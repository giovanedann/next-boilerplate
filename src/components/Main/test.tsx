import { render, screen } from '@testing-library/react'
import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', {
        level: 1,
        name: /boilerplate react \+ nextjs/i
      })
    ).toBeInTheDocument()
  })
})
