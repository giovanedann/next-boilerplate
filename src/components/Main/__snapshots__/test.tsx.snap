// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the background and text colors correctly 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<div>
  <main
    class="c0"
  >
    <img
      alt="Athom image with a Boilerplate text"
      class="c1"
      src="img/logo.svg"
    />
    <h1
      class="c2"
    >
      Boilerplate React + NextJS
    </h1>
    <h2
      class="c3"
    >
       
      TypeScript, React, NextJS, Jest, Testing Library, Storybook and Styled Components
       
    </h2>
    <img
      alt="Developer coding"
      class="c4"
      src="img/hero-illustration.svg"
    />
  </main>
</div>
`
